var/spack/repos/builtin/packages/libxsmm/package.py

```diff
@@ -86,6 +90,9 @@ def install(self, spec, prefix):
         # include symbols by default
         make_args = ['SYM=1']
 
+        if '+header-only' in spec and '@1.6.2:' in spec:
```

Comment:
U also need to throw error if header-only is requested for version which doesn't support it.